document/CoreFormat.mdk

```diff
@@ -617,15 +617,58 @@ The tool should report the problem as an error in `configurationNotifications` a
 A `run` object (&sect;[#run]) may contain a property named `rules`
 whose value is a JSON object, each of whose properties represents an analysis rule.
 
-Each property name in the `rules` object shall be the id of an analysis rule.
+Each property value in the `rules` object shall be a `rule` object (&sect;[#rule]).
+
+If there is only one `rule` object with a particular `id` (&sect;[#rule-id]),
+then the property name for that rule object shall be the rule id.
+
+~ IsoNumberedExample { @IsoNumberedExample=1 }
+In this example, two rules have different ids. The property names match the rule ids.
+
+```
+"rules": {
+  "CA1001": {
+    "id": "CA1001",
+    "shortDescription": "Types that own disposable fields should be disposable."
+  },
+  "CA1002": {
+    "id": "CA1002",
+    "shortDescription": "Do not expose generic lists."
+  }
+}
+```
+~
+
+Some tools use the same rule id to refer to multiple distinct (although logically related) rules.
+In that case, the property names for those rule objects shall be distinct, even though the rule ids
+are the same.
+The property names should be clearly related to the rule id.
 
-Each property value in the `rules` object shall be a `rule` object (&sect;[#rule]) whose
-`id` property (&sect;[#rule-id]) is equal to the property name.
+~ IsoNumberedExample
+In this example, two distinct but related rules have the same rule id. The property names are
+distinct, and are clearly related to the rule id.
+
+```
+"rules": {
+  "CA1711.1": {
```

Comment:
Just a nit, I advised Nick to prefer a hyphen separator character. The '.' is too pervasive as a separator in 'real' names. You could imagine using the dot to actually subcategorize checks, i.e., 100.1, 100.2, 100.3. So, use CA1711-1 for the example. Roslyn actually uses CA1711-001 (to allow for string-based sorting). Note that Roslyn also proposes the let the first instance of a duplicated key come in without munging it, so 'CA1711' would be followed by 'CA1711-1' or whatever. Actually making all keys explicit to demonstrate there are collisions is also a reasonable approach.